AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -284,6 +303,60 @@ - (void)deleteNodesOfKind:(NSString *)kind atIndexPaths:(NSArray *)indexPaths co
   }];
 }
 
+/**
+ * TODO: move
+ */
+- (void)prepareMoveItemChanges:(NSArray *)items
+{
+  dispatch_group_wait(_editingTransactionGroup, DISPATCH_TIME_FOREVER);
+  
+  NSLog(@">>>>> Prepare Item Change"); // FIXME
+  
+  dispatch_group_async(_editingTransactionGroup, _editingTransactionQueue, ^{
+    NSLog(@">>>>> Prepare Item Change - Start."); // FIXME
+    
+    _moveFromIndexPaths = [[NSMutableArray alloc] initWithCapacity:items.count];
+    _moveToIndexPaths = [[NSMutableArray alloc] initWithCapacity:items.count];
+    NSMutableDictionary *_moveIndexPathPairs = [NSMutableDictionary dictionaryWithCapacity:items.count];
+    _moveToNodeContextsDict = [NSMutableDictionary dictionaryWithCapacity:items.count];
+    
+    for (_ASHierarchyMoveItemChange *change in items) {
+      [_moveFromIndexPaths addObject:change.fromIndexPath];
+      [_moveToIndexPaths addObject:change.toIndexPath];
+      [_moveIndexPathPairs setObject:change.toIndexPath forKey:change.fromIndexPath];
+    }
+    
+    for (int i = 0; i < items.count; i++) {
+      NSIndexPath *toIndexPath = [_moveIndexPathPairs objectForKey:_moveFromIndexPaths[i]];
```

Comment:
We can combine this with the for-loop above, I believe, and reduce duplicate accesses.